### PR TITLE
Feat/gcloud region

### PR DESCRIPTION
**Pull request by atbore-phx**
This PR allows to discover DBs worldwide and not only in the pod's region

### CODE REVIEW

**Comment by n-oden** [comment]
Hey, thanks for the PR! I'll try to review this properly before the end of the week.

File: README.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ instance no matter what.  This is configureable at deploy time as
 
 Old Version | New Version | Upgrade Guide
 --- | --- | ---
-v1.0.X | v1.1.0 | [link](UPGRADE.md#v10x--v110)
+v1.1.0 | v1.1.1 | [link](UPGRADE.md#v110--v111)
```
Comment (n-oden) [inline README.md:51]: Please add the 1.1.1 version on top of the 1.0.x version rather than deleting it

File: UPGRADE.md [hunk 2]
```diff
@@ -1,13 +1,5 @@
 # Upgrading Steps
 
-## `v1.0.X` → `v1.1.0`
+## `v1.1.0` → `v1.1.1`
```
Comment (n-oden) [inline UPGRADE.md:3]: Please leave the 1.0.x -> 1.1.1 upgrade steps in place and put the 1.1.0 -> 1.1.1 upgrade instructions above it

File: bin/discovery.sh [hunk 3]
```diff
@@ -61,7 +61,7 @@ while true; do
     gcloud \
       --project "${PROJECT_ID}" \
       sql instances list \
-      --filter "region:${REGION} AND name~^${PRIMARY_INSTANCE_PREFIX} AND state:RUNNABLE AND instanceType:CLOUD_SQL_INSTANCE" \
+      --filter "name~^${PRIMARY_INSTANCE_PREFIX} AND state:RUNNABLE AND instanceType:CLOUD_SQL_INSTANCE" \
```
Comment (n-oden) [inline bin/discovery.sh]: this should be optional behavior; see longer comment above

File: UPGRADE.md [hunk 4]
```diff
@@ -1,13 +1,5 @@
...
-> 🛑 this release changes the default error handling behavior for the metrics/monitoring containers; you will need to update your values.yaml if you wish to preserve the previous behavior.
-
-### Feature highlights
-
-* Added the ability to control pod restart behavior if either of the monitoring containers ([pgpool2\_exporter](https://github.com/pgpool/pgpool2_exporter) and [telegraf](https://github.com/influxdata/telegraf) restart -- depending on your workload, it may not be helpful to interrupt in-flight db transactions due to errors that are outside the critical path
-
-### VALUES - New:
-- `exporter.exitOnError` -- Exit the container if the exporter process exits (otherwise, restart the exporter after a 1s delay); default is false
-- `telegraf.exitOnError` -- Exit the container if the telegraf process exits (otherwise, restart telegraf after a 1s delay); default is false
+> 🛑 this release allows to discover databases instance worldwide inside your Project and not only in the pod region.
```
Comment (n-oden) [inline UPGRADE.md]: nit for grammar/usage: "this release allows discovering database instances worldwide inside your GCP project, and not only in the pod region"

**Review by n-oden — CHANGES_REQUESTED** [review]
I think this is a reasonable thing to want as a feature, but it should probably not be the default, and definitely the default behavior should not change in a patch release.  

Please add a boolean helm configuration value `.discovery.stay_in_region` with the default being `true` so that the default behavior does not change, and make the discovery script construct the filters based on that value.

**Comment by atbore-phx** [comment]
> Please add a boolean helm configuration value .discovery.stay_in_region

Hi @n-oden,
Thanks for the fast reply,
Make this feature optional is the right way to proceed indeed.
Let me have a couple of days and it will be done 👍.

**Comment by atbore-phx** [comment]
hi @n-oden, ready for review ;)

File: bin/discovery.sh [hunk 5]
```diff
@@ -32,6 +32,10 @@ if [ -z "${PRIMARY_INSTANCE_PREFIX}" ]; then
   log fatal "PRIMARY_INSTANCE_PREFIX unset"
 fi
 
+if [ "${STAY_IN_REGION:-}" = "false" ]; then
```
Comment (n-oden) [inline bin/discovery.sh]: A tiny nit, but for consistency please put the default-handing code up in the block of variable declarations in L17-22:

```
STAY_IN_REGION="${STAY_IN_REGION:-"true"}"
```

And then here just say `if ["${STAY_IN_REGION}" = "false" ]...`

File: charts/pgpool-cloudsql/templates/deployment.yaml [hunk 6]
```diff
@@ -83,6 +83,8 @@ spec:
         env:
         - name: PRIMARY_INSTANCE_PREFIX
           value: {{ required "You must set .discovery.primaryInstancePrefix!" .Values.discovery.primaryInstancePrefix | quote }}
+        - name: STAY_IN_REGION
+          value: {{ .Values.discovery.stay_in_region }}
```
Comment (n-oden) [inline charts/pgpool-cloudsql/templates/deployment.yaml]: this needs to be quoted:

```
- name: STAY_IN_REGION
  value: {{ .Values.discovery.stayInRegion | quote }}
```

File: charts/pgpool-cloudsql/values.yaml [hunk 7]
```diff
@@ -63,7 +63,10 @@ discovery:
   # `--filter name:${primaryInstancePrefix}` option and MUST match one and only one
   # non-replica instance.
   primaryInstancePrefix: ""  # REQUIRED!
-
+  # The Google Cloud region where discovery logic looks for databases.
+  # if "true" it's  the same pod's region,
+  # all regions otherwise.
+  stay_in_region: true
```
Comment (n-oden) [inline charts/pgpool-cloudsql/values.yaml]: nit: for consistency here, please use camelCase: 

```
stayInRegion: true
```

File: UPGRADE.md [hunk 8]
```diff
@@ -1,5 +1,9 @@
 # Upgrading Steps
 
+## `v1.1.0` → `v1.1.1`
+
+> 🛑 this release allows discovering database instances worldwide inside your GCP project, and not only in the pod region.
```
Comment (n-oden) [inline UPGRADE.md]: Please add a `### VALUES- new" section covering the new value option; see the v1.1.0 upgrade docs for an example.

Also, change the 🛑 emoji to an ℹ️ emoji, as the former is meant to denote breaking changes.

**Review by n-oden — CHANGES_REQUESTED** [review]
The logic mostly LGTM.  Just a few nits about documentation and style.

Also please address the failing lint test: you need to update the version number in Chart.yaml.

(Don't worry about the failing docker build test; that's unrelated to this PR.)

**Comment by atbore-phx** [comment]
> The logic mostly LGTM. Just a few nits about documentation and style.
> 
> Also please address the failing lint test: you need to update the version number in Chart.yaml.
> 
> (Don't worry about the failing docker build test; that's unrelated to this PR.)

ready for review ;)

**Review by n-oden — APPROVED** [review]
Approved. I'm currently on an airplane but will try to get this merged and published tomorrow.

**Comment by n-oden** [comment]
v1.1.1 is released! Thank you for the contribution!

**Pull request merged**